arch/risc-v/src/mpfs/mpfs_irq.c

```diff
@@ -238,22 +192,11 @@ void up_enable_irq(int irq)
 
       /* Set enable bit for the irq */
 
-      uint64_t hart_id = READ_CSR(mhartid);
-      uintptr_t miebase;
-
-      if (hart_id == 0)
-        {
-          miebase = MPFS_PLIC_H0_MIE0;
-        }
-      else
-        {
-          miebase = MPFS_PLIC_H1_MIE0 +
-            ((hart_id - 1) * MPFS_HART_MIE_OFFSET);
-        }
+      uintptr_t iebase = mpfs_plic_get_iebase();
 
       if (0 <= extirq && extirq <= NR_IRQS - MPFS_IRQ_EXT_START)
         {
-          modifyreg32(miebase + (4 * (extirq / 32)), 0, 1 << (extirq % 32));
+          modifyreg32(iebase + (4 * (extirq / 32)), 0, 1 << (extirq % 32));
```

Comment:
```suggestion
          modifyreg32(iebase + (extirq / 32), 0, 1 << (extirq % 32));
```
together with `uint32_t *iebase = (uint32_t *)mpfs_plic_get_iebase();`

Reply:
I'll modify the init function